core/src/main/scala/kafka/server/metadata/BrokerMetadataListener.scala

```diff
@@ -322,16 +322,22 @@ class BrokerMetadataListener(
 
   private def publish(publisher: MetadataPublisher): Unit = {
     val delta = _delta
-    _image = _delta.apply()
-    _delta = new MetadataDelta(_image)
-    if (isDebugEnabled) {
-      debug(s"Publishing new metadata delta $delta at offset ${_image.highestOffsetAndEpoch().offset}.")
-    }
-    publisher.publish(delta, _image)
+    try {
+      _image = _delta.apply()
+      _delta = new MetadataDelta(_image)
+      if (isDebugEnabled) {
+        debug(s"Publishing new metadata delta $delta at offset ${_image.highestOffsetAndEpoch().offset}.")
+      }
 
-    // Update the metrics since the publisher handled the lastest image
-    brokerMetrics.lastAppliedRecordOffset.set(_highestOffset)
-    brokerMetrics.lastAppliedRecordTimestamp.set(_highestTimestamp)
+      // This publish call is done with its own try-catch and fault handler
+      publisher.publish(delta, _image)
+
+      // Update the metrics since the publisher handled the lastest image
+      brokerMetrics.lastAppliedRecordOffset.set(_highestOffset)
+      brokerMetrics.lastAppliedRecordTimestamp.set(_highestTimestamp)
+    } catch {
```

Comment:
can you please scope this to just delta.apply? We already have other exceptions thrown here and we don't want to double count faults.